### PR TITLE
Introduce StoredBoolean

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/UnixPath.java [hunk 1]
```diff
@@ -50,10 +52,24 @@ public Path toPath() {
...
         return new String(readBytes(), StandardCharsets.UTF_8);
     }
 
+    public Optional<String> readUtf8FileIfExists() {
+        try {
+            return Optional.of(new String(Files.readAllBytes(path), StandardCharsets.UTF_8));
+        } catch (NoSuchFileException ignored) {
+            return Optional.empty();
+        } catch (IOException e) {
+            throw new RuntimeIOException(e);
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/UnixPath.java]: `java.io.UncheckedIOException`?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/StoredBoolean.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    public StoredBoolean(Path path) {
+        this.path = new UnixPath(path);
+    }
+
+    public boolean value() {
+        return path.readUtf8FileIfExists().map(String::trim).map(s -> !"0".equals(s)).orElse(false);
+    }
+
+    /** Sets value to true. */
+    public void set(TaskContext context) {
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/StoredBoolean.java]: Add a method that can sets it equal to the given `boolean` instead/also?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/UnixPath.java (hunk 1 above)
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/UnixPath.java]: Yes!

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/StoredBoolean.java (hunk 2 above)
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/StoredBoolean.java]: ok, also

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/StoredBoolean.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        if (!value()) {
+            context.log(logger, "Writes " + path);
+            path.writeUtf8File("1");
+        }
+    }
+
+    /** Sets value to true. */
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/StoredBoolean.java]: :neutral_face:

Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/StoredBoolean.java]: doh

**Review by freva — APPROVED** [review]
:100:

**Pull request merged**